### PR TITLE
[CORE-390] Add initial load compaction module

### PR DESCRIPTION
**Pull request by TelicentPaul**
This will ensure that the on-disk TDB2 will be compacted at server start-up. It can be disabled via environment variable should we wish to. 

Also, some minor code tidy-up. Removing some unused imports and some Commons Collection code that can be easily replaced with core code.

### CODE REVIEW

File: scg-system/src/main/java/io/telicent/core/FMod_InitialCompaction.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    /**
+     * Check the given Graph and, if possible, return the underlying TDB2 instance
+     * @param dsg Graph
+     * @return TDB2 compatible DSG or null
+     */
+    public static DatasetGraph getTDB2(DatasetGraph dsg) {
+        for (; ;) {
+            if (IS_NOT_TDB_2.test(dsg))
```
Comment (rvesse) [inline scg-system/src/main/java/io/telicent/core/FMod_InitialCompaction.java:78]: I'm not sure this test is necessary/wanted here.  If we have a `DatasetGraphABAC` configured for example then it's going to fail this test and bail out of the loop and not trigger a compact as a result

I think we just want the two other tests, then we keep unwrapping until we either find a TDB2 dataset or we don't

File: scg-system/src/test/java/io/telicent/TestInitialCompaction.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    public void test_persistentDataset() {
+        // given
+        mockDatabaseMgr.when(() -> DatabaseMgr.compact(any(),anyBoolean()))
+                .thenAnswer((Answer<Void>) invocation -> null);
+        String configFile = "config-persistent.ttl";
+        // when
+        server = launchServer(configFile);
+        // then
+        assertNotNull(server.serverURL());
+        mockDatabaseMgr.verify(() -> DatabaseMgr.compact(any(), anyBoolean()), times(2));
```
Comment (rvesse) [inline scg-system/src/test/java/io/telicent/TestInitialCompaction.java:83]: Why `times(2)` here?

File: scg-system/src/test/java/io/telicent/TestInitialCompaction.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        // given
+        FMod_InitialCompaction fModInitialCompaction = new FMod_InitialCompaction();
+        // when
+        String name = fModInitialCompaction.name();
+        // then
+        assertNotNull(name);
+    }
+
+    @Test
+    public void test_TDB1_DSG() {
```
Comment (rvesse) [inline scg-system/src/test/java/io/telicent/TestInitialCompaction.java:97]: Could maybe do with a few more test cases like this that test that different datasets we commonly expect e.g. `DatasetGraphABAC` are correctly unwrapped for compaction when backed by TDB2?

File: scg-system/src/main/java/io/telicent/core/FMod_InitialCompaction.java (hunk 1 above)
Comment (TelicentPaul) [reply scg-system/src/main/java/io/telicent/core/FMod_InitialCompaction.java:78]: Fair enough

File: scg-system/src/test/java/io/telicent/TestInitialCompaction.java (hunk 3 above)
Comment (TelicentPaul) [reply scg-system/src/test/java/io/telicent/TestInitialCompaction.java:94]: Yeah, just laziness on my part as I could see it was getting called by other tests by the coverage checks. 
I can add them

Comment (TelicentPaul) [reply scg-system/src/test/java/io/telicent/TestInitialCompaction.java:94]: Are there any other common datasets used outside of DatasetGraphABAC?

Comment (rvesse) [reply scg-system/src/test/java/io/telicent/TestInitialCompaction.java:94]: Not for our usage of Jena

**Review by rvesse — APPROVED** [review]
LGTM, guess next step is rolling this onto the system integration cluster and verifying it works in the real environment?

**Comment by TelicentPaul** [comment]
Let's hold off on a release as it'll need bumped when I upgrade the Jena Kafka module with the other fix.

**Comment by TelicentPaul** [comment]
I think our large datasets will be a good test.  Once we have the other complimentary processes in place (nightly compaction) it should be less of a concern.

**Pull request merged**